### PR TITLE
Review credential commands applying to hobby databases and messaging

### PR DESCRIPTION
**Pull request by camillebaldock**
(I'm going to add a couple new tests for some edge cases uncovered while doing this). 

Commands for hobby:

* `heroku pg:credentials:create --name nope -a camille-hobby-app`

`This operation is not supported by Hobby tier databases.`

* `heroku pg:credentials:repair-default -a camille-hobby-app`

`This operation is not supported by Hobby tier databases.`

* `heroku pg:credentials -a camille-hobby-app`

```
Credential  State
──────────  ──────
default     active
```

* `heroku pg:credentials --reset -a camille-hobby-app`

`Resetting credentials on postgresql-polished-42344... done`

* `heroku pg:credentials:url -a camille-hobby-app`

```
Connection information for default credential.
Connection info string:
   "dbname=something host=ec2-23-21-220-23.compute-1.amazonaws.com port=5432 user=something password=something sslmode=require"
Connection URL:
   postgres://something:something@ec2-23-21-220-23.compute-1.amazonaws.com:5432/something
```

* `heroku pg:credentials:url --name nope -a camille-hobby-app`

Currently:
`This operation is not supported by Hobby tier databases.`

Proposal:
`Could not find credential nope for database postgresql-hobby-56874.
Only one default credential is supported for Hobby tier databases.
`

* `heroku pg:credentials:rotate --all -a camille-hobby-app`

`Rotating all credentials on postgresql-polished-42344... done`

* `heroku pg:credentials:rotate --name default -a camille-hobby-app`

```
▸    WARNING: Destructive Action
▸    The password for the default credential will rotate.
▸    Connections will be reset and applications will be restarted.
▸    To proceed, type camille-hobby-app or re-run this
▸    command with --confirm camille-hobby-app

> camille-hobby-app
Rotating default on postgresql-polished-42344... done
```

* `heroku pg:credentials:rotate --name default --force -a camille-hobby-app`

```
▸    WARNING: Destructive Action
▸    The password for the default credential will rotate.
▸    Connections will be reset and applications will be restarted.
▸    To proceed, type camille-hobby-app or re-run this
▸    command with --confirm camille-hobby-app

> camille-hobby-app
Rotating default on postgresql-polished-42344... done
```

Current:
`Rotating default on postgresql-polished-42344... done`

Proposed:
`Cannot force rotate the default credential.`

* `heroku pg:credentials:rotate --name nope -a camille-hobby-app`

Currently:
`This operation is not supported by Hobby tier databases.`

Proposal:
`Could not find credential nope for database postgresql-hobby-56874.
Only one default credential is supported for Hobby tier databases.
`

* `heroku pg:credentials:destroy --name nope -a camille-hobby-app`

Currently:
`This operation is not supported by Hobby tier databases.`

Proposal:
`Could not find credential nope for database postgresql-hobby-56874.
Only one default credential is supported for Hobby tier databases.
`

* `heroku pg:credentials:destroy --name default -a camille-hobby-app`

Currently:
`Not found.`

Proposal:
`Default credential cannot be destroyed.`

### CODE REVIEW

**Review by genslein — APPROVED** [review]
I think this is clearer to the customer what's occurring, curious why codecov dropped in % though? not blocking the PR for it

File: commands/credentials/destroy.js [hunk 1]
```diff
@@ -11,8 +11,15 @@ function * run (context, heroku) {
...
 
+  if (cred === 'default') {
+    throw new Error('Default credential cannot be destroyed.')
+  }
+
   let db = yield fetcher.addon(app, args.database)
-  if (util.starterPlan(db)) throw new Error('This operation is not supported by Hobby tier databases.')
+  if (util.starterPlan(db)) {
+    throw new Error(`Could not find credential ${cred} for database ${db.name}.
+Only one default credential is supported for Hobby tier databases.`)
```
Comment (msakrejda) [inline commands/credentials/destroy.js]: The first part of this feels superfluous. Maybe just `Only the default credential is supported for Hobby tier databases.`?

File: commands/credentials/rotate.js [hunk 2]
```diff
@@ -15,8 +15,13 @@ function * run (context, heroku) {
...
   }
   let cred = flags.name || 'default'
-  if (util.starterPlan(db) && cred !== 'default') throw new Error('This operation is not supported by Hobby tier databases.')
-
+  if (cred === 'default' && 'force' in flags && !all) {
+    cli.exit(1, 'Cannot force rotate the default credential.')
+  }
+  if (util.starterPlan(db) && cred !== 'default') {
+    throw new Error(`Could not find credential ${cred} for database ${db.name}.
+Only one default credential is supported for Hobby tier databases.`)
```
Comment (msakrejda) [inline commands/credentials/rotate.js]: Same as above.

File: commands/credentials/url.js [hunk 3]
```diff
@@ -13,7 +13,10 @@ function * run (context, heroku) {
 
   let db = yield fetcher.addon(app, args.database)
   let cred = flags.name || 'default'
-  if (util.starterPlan(db) && cred !== 'default') throw new Error('This operation is not supported by Hobby tier databases.')
+  if (util.starterPlan(db) && cred !== 'default') {
+    throw new Error(`Could not find credential ${cred} for database ${db.name}.
+Only one default credential is supported for Hobby tier databases.`)
```
Comment (msakrejda) [inline commands/credentials/url.js]: Same as above.

**Review by msakrejda — APPROVED** [review]
Some suggestions, but +1
